dto/api_input.py

```diff
@@ -56,3 +62,30 @@ class GamesDTO(BaseDTO):
     per_page = fields.Int(required=False, missing=0)
 
 
+class InputUserDTO(BaseDTO):
+    username = fields.Str(required=True)
+    email = fields.Str(required=True)
+    password_hash = fields.Str(required=True)
+
+
+class ResetPasswordDTO(BaseDTO):
+    email = fields.Str(required=True)
+
+
+class NewPasswordDTO(BaseDTO):
+    new_password = fields.Str(required=True)
+    email = fields.Str(required=True)
+
+
+class InputUserLoginDTO(BaseDTO):
+    email_or_username = fields.Str(required=True)
+    password_hash = fields.Str(required=True)
+
+
+class InputUserPreferencesDTO(BaseDTO):
+    preferences = fields.List(fields.Int, required=True)
+    user_id = fields.Int(required=False, missing=None)
+
+
+class InputGetUserPreferencesDTO(BaseDTO):
```

Comment:
InputGetUserPreferencesDTO and InputUserPreferencesDTO what's the difference between the two and why do we need both? (I also suggest renaming them to some more specific names. "Get" part is not adding nuch of a context IMO)